File: src/main/java/com/kingmang/ixion/ast/AssignExpression.java
package com.kingmang.ixion.ast;

import com.kingmang.ixion.ExprVisitor;
import com.kingmang.ixion.lexer.Position;

public final class AssignExpression extends Expression {
    public final Expression left;
    public final Expression right;

    public AssignExpression(Position pos, Expression left, Expression right) {
        super(pos);
        this.left = left;
        this.right = right;
    }

    public <R> R accept(ExprVisitor<R> visitor) {
        return visitor.visitAssignExpr(this);
    }
}

File: src/main/java/com/kingmang/ixion/ast/AssignExpression.kt
package com.kingmang.ixion.ast

import com.kingmang.ixion.ExprVisitor
import com.kingmang.ixion.lexer.Position

class AssignExpression(pos: Position?, @JvmField val left: Expression?, @JvmField val right: Expression?) : Expression(pos) {
    override fun <R> accept(visitor: ExprVisitor<R>): R {
        return visitor.visitAssignExpr(this)
    }
}